autorest/adal/token.go

```diff
@@ -365,6 +365,25 @@ func (secret ServicePrincipalAuthorizationCodeSecret) MarshalJSON() ([]byte, err
 	})
 }
 
+// ServicePrincipalFederatedSecret implements ServicePrincipalSecret for Federated JWTs.
+type ServicePrincipalFederatedSecret struct {
+	Jwt string
```

Comment:
I don't think this needs to be exported.

Reply:
Maybe this is intended as a (goroutine unsafe) way to replace JWTs as they expire? Does `adal` have a different pattern for that sort of thing?

Reply:
Good point.  There's no established pattern for this in `adal` today, so if the `Jwt` needs to be updated, then you'd have to export it like at present (it's racy like you said).  Also, if this is the case, then the field should be named `JWT` for consistency.

@chlowell do we need to add support for this in `azidentity`?

Reply:
Or perhaps our current abstractions make it possible for customers to write their own implementation.

Reply:
For `azidentity` this is part of https://github.com/Azure/azure-sdk-for-go/issues/15615. Planning to take a callback for updating the assertion.

Reply:
Do we want applications to create a new `ServicePrincipalFederatedSecret` after the JWT expires?